### PR TITLE
feat(docs): add initial docs dir, start page and why a workspace page

### PR DESCRIPTION
**Pull request by jschwarty**

### CODE REVIEW

File: docs/why-a-workspace.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+and make decisions accordingly. So, here are some of the potential disadvantages that
+we want to be aware of when it comes to a mono repo workspace.
+
+- Takes work to try and limit access to parts of the code base
+- An upgrade to a lib requires a change to all implementors (can't roll out different versions side by side)
+- Can lead to accepted dependencies, making it overkill to work on a small feature
+    - Say some library code is designed to hit the web service, in a single workspace
+      you know the service is available so effort might not be made to be able to run
+      that library code with a mock service
+- The need for build optimization and other tooling
```
Comment (vsavkin) [inline docs/why-a-workspace.md]: I think this need is present regardless of the monorepo. 

monorepo is a way to organize code, it doesn't affect how the code gets bundled and shipped

File: docs/why-a-workspace.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# Nx Workspace
```
Comment (vsavkin) [inline docs/why-a-workspace.md]: I would write a paragraph about how pretty much every large company works this way (Google, FAcebook, Uber, Twitter). It's important to convey that this approach isn't new. It's being tested.

File: docs/readme.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
+# Narwhal Extensions for Angular
```
Comment (vsavkin) [inline docs/readme.md]: This is the steps I follow when I explain Nx to our clients:



Large companies specific requirements when it comes to building software.

1. They don't build small apps. They have multiple teams building multiple apps using multiple shared libs. It's many to many to many. Organizing this dev workflow is challenging.
2. They care about consistency. AngularCLI has some conventions but they don't go far enough. What about state management, what about routing? Even small things code formatting?
3. They have legacy angularjs apps they need to migrate. NgUpgrade is great, but still a lot of work to set up.
4. They want to write robust code: error handling, race conditions.
5. ...



These are, among other, problems that Nx addresses.

1. It gives you Nx Workspace--a monorepo way of building frontend apps. The same approach used by Google, Facebook, Uber, Twitter and many others.
2. It gives you code generation abilities to generate state management stuff and routing. It also comes with runtime libraries to make your code more robust. It comes with a code formatter (used by Google!).
3. It comes with code generation abilities to set up a hybrid app in minutes. So you can take advantage of your existing angularjs code right away.
...



Then I talk about each of those parts separately. And I cover the two node modules separately.

Basically I try to explain the problems people have first, before going into what schematics are. What do you think?

**Review by vsavkin — CHANGES_REQUESTED** [review]

File: docs/readme.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+providing file/folder structure for enterprise Angular development as well as providing generator recipes for
+creating new code in your applications. The libraries provide a set of code that you can import and use 
+to simplify common patterns and functionality in your Angular applications.
+
+#### Getting Started
+```npm install --save @nrwl/nx```
+
+or
+
+```yarn add @nrwl/nx```
```
[the remark this answers is not in the document]
Comment (jschwarty) [reply docs/readme.md]: Updated

File: docs/readme.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+That approach begins with the concept of a workspace, or what we call a **Nx Workspace**.
+
+## Nx Workspace
+This workspace contains your Angular application (or applications) and any supporting libraries you create. It is a mono
+repo for your application domain.
+
+[Why a workspace (or mono repo) is needed](why-a-workspace.md)
+
+The AngularCLI can actually be set up manually to support multiple Angular applications and referenced library
```
[the remark this answers is not in the document]
Comment (jschwarty) [reply docs/readme.md]: Added example of running `ng new myprojectname --collection=@nrwl/schematics`.

File: docs/why-a-workspace.md (hunk 2 above)
Comment (jschwarty) [reply docs/why-a-workspace.md]: Added

File: docs/why-a-workspace.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+## Why is a workspace needed?
+
+Let's take, for example, an application domain for a support ticket system which would
+would consist of:
+- Customer application for submitting tickets (Angular app)
+- Admin application for working tickets (Angular app)
+- Library of shared code for the above (TypeScript)
```
[the remark this answers is not in the document]
Comment (jschwarty) [reply docs/why-a-workspace.md]: changed

File: docs/why-a-workspace.md (hunk 1 above)
Comment (jschwarty) [reply docs/why-a-workspace.md]: Removed that bullet point

File: docs/readme.md (hunk 3 above)
Comment (jschwarty) [reply docs/readme.md]: Added a "why Nx" section with this info.

File: docs/readme.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+```
+It's similar to the standard CLI projects with a few changes:
+- We have the `apps` dir where all applications are placed
+- We have the `libs` dir where all custom library code is placed
+- Some files have moved to the root: `tsconfigs, test.js`, so all apps and libs share them.
+
+All of the standard AngularCLI commands are available, again because **Nx** simply extends the power of the
+AngularCLI.
+
+You can build/serve/test any of the apps by passing the app name.
```
Comment (vsavkin) [inline docs/readme.md]: I would not talk about `build` here because there is nothing in an empty workspace that can be built.

Instead, I'd talk about generating apps and libs here. How to generate an app, how to generate a lib. What is an app? What is a lib? How an app can depend on a lib. 

Something like that:

```
## Apps and Libs

* App is something we actually ship to the user (it is a bundle or a group of bundles). It's akin to a binary.
* Libs is something we compose apps from. We don't ship libs to the user.

This is how you generate app: ... I would merely mention schematics here with a link to a maybe footnote.
This is how you generate lib: ...
This is how your app or or your lib can depend on another lib (mention npmscope)

Here we can link link to a different doc with in depth explanation of apps, libs and best practices: all the command line options, what should go into apps, how to structure libs, how to make sure lazy loading works. 

And here I would show how to build/test/serve apps and libs. 
```

With this in place, the workspace stuff should be covered, and we can talking about other things Nx provides.  Something like

```
Nx helps you have consistent code.

* Check out our schematics and runtime libraries for state management (Ngrx) and communicating with the server (link)
* Check out our schematics for upgrade (link)
* Check out our code formatter and lint checks (link)
...
```

What do you think?

File: docs/why-a-workspace.md (hunk 2 above)
Comment (vsavkin) [reply docs/why-a-workspace.md]: I like it
